gr-uhd/apps/uhd_siggen_base.py

```diff
@@ -55,6 +57,12 @@ class USRPSiggen(gr.top_block, pubsub, UHDApp):
     def __init__(self, args):
         gr.top_block.__init__(self)
         pubsub.__init__(self)
+        # If the power argument is given, we need to turn that into a power
+        # *reference* level. This is a bit of a hack because we're assuming
+        # knowledge of UHDApp (i.e. we're leaking abstractions). But it's simple
+        # and harmless enough.
+        if args.power:
+            args.power -= 20 * math.log10(args.amplitude)
```

Comment:
should there be a check here for `args.amplitude > 0.0`?

Reply:
Yeah good idea. It should be elsewhere, I guess, since a negative amplitude makes no sense anyway.

Reply:
I've put it right before this.